tests/test_extract.py

```diff
@@ -1,49 +1,141 @@
+def test_read_tsv_into_df():
+    with patch.object(pd, "read_csv", return_value=pd.DataFrame()) as patch_pd_read_csv:
+        df = extract.read_tsv_into_df(tsv_path="./tests/test_assets/test_file.tsv")
+        patch_pd_read_csv.assert_called_once_with(
+            "./tests/test_assets/test_file.tsv", sep="\t"
+        )
+        assert isinstance(df, pd.DataFrame)
+
+
+def test_read_table_into_df(syn):
+    mock_df = MockAsDF()
+    with patch.object(syn, "tableQuery", return_value=mock_df) as patch_syn_tablequery:
+        df = extract.read_table_into_df(table_id="syn11111111", syn=syn)
+        patch_syn_tablequery.assert_called_once_with("select * from syn11111111")
+        assert isinstance(df, pd.DataFrame)
+        assert df.equals(mock_df.asDataFrame())
+
+
+def test_read_feather_into_df_if_not_feather():
+    with pytest.raises(ValueError, match="Please make sure *"):
+        extract.read_feather_into_df(feather_path="./tests/test_assets/test_file.csv")
+
+
+def test_read_feather_into_df():
+    with patch.object(
+        pd, "read_feather", return_value=pd.DataFrame()
+    ) as patch_pd_read_feather:
+        df = extract.read_feather_into_df(
+            feather_path="./tests/test_assets/test_file.feather"
+        )
+        patch_pd_read_feather.assert_called_once_with(
+            "./tests/test_assets/test_file.feather"
+        )
+        assert isinstance(df, pd.DataFrame)
+
 
-def test_get_entity_as_df():
-    assert type(extract.get_entity_as_df(syn_id="syn25838562", format="csv")) is DataFrame
-    with pytest.raises(SystemExit) as err:
-        extract.get_entity_as_df(syn_id="syn25838562", format="tsv")
-    assert err.type == SystemExit
-    assert err.value.code == 9
+def test_read_json_into_df_if_not_json():
+    with pytest.raises(ValueError, match="Please make sure *"):
+        extract.read_json_into_df(json_path="./tests/test_assets/test_file.csv")
 
-    assert type(extract.get_entity_as_df(syn_id="syn25838563", format="tsv")) is DataFrame
-    with pytest.raises(SystemExit) as err:
-        extract.get_entity_as_df(syn_id="syn25838563", format="csv")
-    assert err.type == SystemExit
-    assert err.value.code == 9
 
-    assert type(extract.get_entity_as_df(syn_id="syn25838546", format="table")) is DataFrame
-    with pytest.raises(SystemExit) as err:
-        extract.get_entity_as_df(syn_id="syn25838563", format="table")
-    assert err.type == SystemExit
-    assert err.value.code == 1
+def test_read_json_into_df():
+    with patch.object(
+        pd, "read_json", return_value=pd.DataFrame()
+    ) as patch_pd_read_json:
+        df = extract.read_json_into_df(json_path="./tests/test_assets/test_file.json")
+        patch_pd_read_json.assert_called_once_with(
+            "./tests/test_assets/test_file.json", orient="records"
+        )
+        assert isinstance(df, pd.DataFrame)
+
+
+# test if utils._login_to_synapse is called when syn=None
+def test_get_entity_as_df_syn_is_none(syn):
```

Comment:
Does this work?  You're not patching `read_table_into_df`

Reply:
This does work the way that I intended, here I am only testing if the `_login_to_synapse` function is called when `syn=None`. Because the mocked `syn` object has the `get` method, it passes. if I was also testing that the result of `extract.get_entity_as_df` is still a dataframe when `syn=None` I imagine I would need to mock  `read_table_into_df` as well. Should I be doing that?

Reply:
Actually - I think its fine because you've tested that functionality elsewhere.  Thanks!